### PR TITLE
AWS Remote Contol (WINDOWS CLIENT, PYTHON)

### PR DESCRIPTION
**Pull request by TheDevFreak**
Windows python client for #14 

Supports the messagebox action and the showimage action (complete coverage at the moment)

### CODE REVIEW

**Comment by TheDevFreak** [comment]
Yes, can do. I'll add it now.

**Comment by TheDevFreak** [comment]
I have now added a config file.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+with open("config.json") as conf:
+    jconf = json.load(conf)
+    API_BASE = jconf['API_BASE']
+    API_KEY = jconf['API_KEY']
+
+#Different Types Of Boxes
+#Ok Only
+MB_OK = 0x0
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: No?

I don't really know how, if you could give any pointers that would be great.

All i can find would be:
```
from enum import Enum
class Types(Enum):
    MB_OK = 0x0
    MB_OKCXL = 0x01
```

then access those with:
`Types.<whatever>`

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    choices = [list(string.ascii_letters), list(string.digits)]
+    for i in range(0, random.randint(12,14)):
+        cID = cID + random.choice(random.choice(choices))
+    Mbox("Client ID", cID, MB_OK, ICON_INFO)
+    return cID
+
+cID = generateClientID()
+
+def runner(cID):
+    with urllib.request.urlopen(API_BASE+cID+"?key="+API_KEY) as response:
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Probably a good idea.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+bicons_map = {"exclaim": ICON_EXCLAIM, "info": ICON_INFO, "question": ICON_QUESTION, "stop": ICON_STOP}
+
+def Mbox(title, content, type=MB_OK, icon=None):
+    if icon == None:
+        result = ctypes.windll.user32.MessageBoxW(0, content, title, type)
+    else:
+        result = ctypes.windll.user32.MessageBoxW(0, content, title, type | icon)
+    return result
+
+def generateClientID():
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Looks good to me.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+#Written for Python 3.6.x
+
+import urllib.request
```
[the remark this answers is not in the document]
Comment (yukihyou) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Requests is _big_ too, I avoided using it when I was messing with some stuff on AWS - simple built-ins are sufficient for most things.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+#Exclamation Point
+ICON_EXCLAIM = 0x30
+#Lowercase i in circle
+ICON_INFO = 0x40
+#Question Mark
+ICON_QUESTION = 0x20
+#"Stop" Cross
+ICON_STOP = 0x10
+
+btypes_map = {"okonly": MB_OK, "okcancel": MB_OKCXL, "yesnocancel": MB_YESNOCXL, "yesno": MB_YESNO}
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Not sure, the maps are used to easily access from the api.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    with urllib.request.urlopen(API_BASE+cID+"?key="+API_KEY) as response:
+        jsonR = response.read()
+        jsonR = json.loads(jsonR)
+        return jsonR
+
+#Mbox("Wirus Alert", "I am wirus push ok to accept wirus to you network. you need fixation network security", MB_OK, ICON_EXCLAIM)
+
+while True:
+    result = runner(cID)
+    for act in result:
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: When you POST the api you have to provide a clientID iirc @yukihyou can clear this up.

[the remark this answers is not in the document]
Comment (yukihyou) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: @PixeLInc Hit me up on Discord and I'll go over the basics of the API itself - ClientID is provided by the client, and simply echoed back as part of the reply - it's used on the server side to filter requests.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (PixeLInc) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Okay, I guess that works.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PixeLInc) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: Yeah, generally that's what you'd do.

Not like a big deal, but just makes it look nicer :P

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (PixeLInc) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: eh - should be good for now.

File: remote-control/aws-based-rc/client/PythonClient/PythonClient.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (PixeLInc) [reply remote-control/aws-based-rc/client/PythonClient/PythonClient.py]: It's all good. I looked over it a little. It'll do for now ;)

**Comment by llamositopia** [comment]
Once this branch gets rebased to the current master, and sorted properly, I'll merge in

**Comment by TheDevFreak** [comment]
I don't really know how to rebase, but I've tried!

**Pull request merged**